src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/api/impl/Channel.kt

```diff
@@ -164,18 +165,28 @@ class Channel(
             val event = if (mode.mangle) mangler.onOutgoing(this@Channel, buffer, metadata) else null
             val messageId = nextMessageId(bookName, sessionGroup, sessionAlias, SECOND)
 
+            val data = Unpooled.copiedBuffer(buffer) // copied because buffer will be released after sending
```

Comment:
I think we also need to add a note that the processing of the `data` is done asynchronously and this is why we need to preserve the buffer (it might be released before we start processing by mangler and onMessage handler)